via/static/scripts/video_player/hooks/use-app-layout.ts

```diff
@@ -1,24 +1,41 @@
 import type { RefObject } from 'preact';
-import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+import { useCallback, useLayoutEffect, useRef, useState } from 'preact/hooks';
 
 import { SIDEBYSIDE_THRESHOLD, TOOLBAR_WIDTH } from './use-side-by-side-layout';
 
 /* Relative application container size */
 export type AppSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
 
+export type AppLayoutInfo = {
+  /* Relative size of app container element */
+  appSize: AppSize;
+  multicolumn: boolean;
+  /* Width for the column containing the transcript as a CSS dimension */
+  transcriptWidth: string;
+};
+
+const transcriptWidths = {
+  sm: '100%',
+  md: '380px',
+  lg: '410px',
+  xl: '450px',
+};
+
 /**
- * Return a string representing the relative size of `appContainer` that can
- * be used to make decisions about app layout and sizing of components. This
- * would typically be something handled by media queries, but it is an element
- * (`body`) width that needs to be queried, not the viewport. We could replace
- * this with container queries once browser support is sufficient.
+ * Return app layout info based on the size of the `appContainer` element. We
+ * could replace this with container queries once browser support is sufficient.
  *
  * NB: This hook will not update if `appContainer.current` changes.
  */
 export function useAppLayout(
   appContainer: RefObject<HTMLDivElement | null>
-): AppSize {
-  const [appSize, setAppSize] = useState<AppSize>('sm');
+): AppLayoutInfo {
+  const lastContainerSize = useRef<AppSize>('sm');
+  const [layoutInfo, setLayoutInfo] = useState<AppLayoutInfo>({
+    appSize: 'sm',
+    multicolumn: false,
+    transcriptWidth: transcriptWidths.sm,
```

Comment:
`multicolumn` and `transcriptWidth` can be computed from the `appSize` value, right? In that case you could also store just the `appSize` in state and compute the other properties from that.